### PR TITLE
feat: Preserve ID3 tags and cover art in FLAC conversions using FFmpeg

### PR DESCRIPTION
**Pull request by Ardakilic**
This pull request introduces support for preserving ID3 tags and cover art from original FLAC files during conversion to 16-bit FLAC files. The preservation is handled using FFmpeg by default, with an option to disable it via the `--no-preserve-metadata` flag. This enhances the tool's ability to maintain metadata integrity without requiring manual post-processing.

Key improvements:
- **Metadata Preservation**: Automatically copies ID3 tags and embedded cover art to the converted FLAC files using FFmpeg.
- **Fallback Mechanism**: If FFmpeg is unavailable or preservation is disabled, the tool falls back to simple file copying or renaming to ensure conversion proceeds.
- **New Configuration Flag**: Added `--no-preserve-metadata` to opt-out of metadata preservation for scenarios where it's not needed.
- **FFmpeg Dependency Check**: Validates FFmpeg installation in local mode when preservation is enabled, providing clear error messages.
- **Updated Documentation**: README.md now includes FFmpeg installation instructions and updated usage examples reflecting the new feature.
- **Comprehensive Testing**: Added unit tests in `main_test.go` to cover metadata preservation success/failure cases, Docker mode, edge cases like non-audio files, and integration with existing conversion logic.
- **gitignore Updates**: Added coverage output files to ignore patterns for cleaner repository state.

The feature is backward-compatible and does not alter existing behavior when the flag is not used. It supports both local SoX/FFmpeg execution and Docker mode (with appropriate image support for FFmpeg).

### Files Modified
- **.gitignore**: Added `coverage.out` and `coverage.html` to exclude test coverage outputs.
- **README.md**: 
  - Added bullet point for metadata preservation in feature list.
  - Included FFmpeg installation instructions for various OS.
  - Added `--no-preserve-metadata` to usage flags.
  - Updated step-by-step conversion process to include metadata preservation.
- **main.go**:
  - Added `NoPreserveMetadata` field to `Config` struct.
  - Registered new flag `--no-preserve-metadata`.
  - Added FFmpeg availability check in `setupSoxCommand()` if preservation is enabled.
  - Integrated metadata merging logic using FFmpeg in conversion pipeline (full implementation in the omitted lines).
- **main_test.go**:
  - New test `TestProcessAudioFiles()` for directory processing, subdirectory handling, fallback copying, and non-audio file skipping.
  - New test `TestConvertFlac()` with subtests for no-conversion copy, bitrate/sample rate conversions, metadata preservation success/failure, Docker mode, and SoX failure fallback.
  - New test `TestSelfUpdateFullFlow()` for end-to-end self-update with mocked API and archive handling.
  - New test `TestSelfUpdateAPI404()` for handling 404 responses from GitHub API.
  - New test `TestSelfUpdateInvalidArchive()` for extraction failures.
  - New test `TestRunConverterEdge()` for target directory creation failures, self-update with args, and image copying behavior.
  - New test `TestMergeMetadataDocker()` for Docker-based metadata merging.

## Testing
- All new tests pass locally.
- Verified with sample FLAC files containing tags and cover art; metadata is preserved post-conversion.
- Tested fallback scenarios (no FFmpeg, disabled flag) ensure no breakage.
- Docker integration tested with appropriate images.

## Checklist
- [x] Code changes implemented
- [x] Tests added/updated
- [x] Documentation updated
- [x] No breaking changes
- [x] Verified with local and Docker execution

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -283,45 +292,161 @@ func determineConversion(info *AudioInfo) (bool, []string, []string) {
...
+	if !config.NoPreserveMetadata {
+		// Merge metadata using FFmpeg
+		if mergeErr := mergeMetadataWithFFmpeg(sourcePath, tempPath, targetPath); mergeErr != nil {
+			fmt.Printf("Warning: Metadata preservation failed for %s, keeping converted audio without tags: %v\n", targetPath, mergeErr)
+			// Fallback: rename temp to target
+			if renameErr := os.Rename(tempPath, targetPath); renameErr != nil {
+				return fmt.Errorf("fallback rename failed after metadata merge error: %w", renameErr)
+			}
+			return nil
+		}
```
[the remark this answers is not in the document]
Comment (Ardakilic) [reply main.go:353]: sox transcoding doesn't include any tags.
